pkg/collector/gadgets/dns_tracer_collector.go

```diff
@@ -0,0 +1,96 @@
+package gadgets
+
+import (
+	"fmt"
+	"github.com/cilium/ebpf/rlimit"
+	"log"
+	"time"
+
+	"github.com/Azure/aks-periscope/pkg/interfaces"
+	"github.com/Azure/aks-periscope/pkg/utils"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/dns/tracer"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/dns/types"
+	"os"
+)
+
+// DNSTracerCollector defines a DNS tracer Collector struct
+type DNSTracerCollector struct {
+	data         map[string]string
+	osIdentifier utils.OSIdentifier
+}
+
+// NewDNSTracerCollector is a constructor to collect DNS trace data using IG
+func NewDNSTracerCollector(osIdentifier utils.OSIdentifier) *DNSTracerCollector {
+	return &DNSTracerCollector{
+		osIdentifier: osIdentifier,
+		data:         make(map[string]string),
+	}
+}
+
+func (collector *DNSTracerCollector) GetName() string {
+	return "dns-tracer"
+}
+
+func (collector *DNSTracerCollector) CheckSupported() error {
+	// check for OS since ebpf is only available on linux
+	if collector.osIdentifier != "linux" {
```

Comment:
💡 please refer to this we could use `utils` for `linux or windows`, for conditional check instead of string comparison: https://github.com/Azure/aks-periscope/blob/master/pkg/collector/dns_collector.go#L39 


```*.go
if collector.osIdentifier != utils.Linux { ... }
```

Reply:
1. the PR is raised against the go1.18 branch so that the files changed in this feature are correctly highlighted. the intention is to merge the Go version update PR then we can adjust the base branch of this branch accordingly. sorry about the confusion

Reply:
2. as discussed, I have forked the repo and am going to follow the proper workflow process. thanks @Tatsinnit for your generous help

Reply:
> 💡 please refer to this we could use `utils` for `linux or windows`, for conditional check instead of string comparison: https://github.com/Azure/aks-periscope/blob/master/pkg/collector/dns_collector.go#L39
> 
> ```go
> if collector.osIdentifier != utils.Linux { ... }
> ```

well spotted! updated.